Microsoft.Toolkit.Uwp.UI.Controls.Design/Common/Constants.cs

```diff
@@ -4,12 +4,18 @@
 
 using System;
 using System.Diagnostics.CodeAnalysis;
-using System.Linq.Expressions;
 
 [assembly: SuppressMessage("General", "SWC1001:XmlDocumentationCommentShouldBeSpelledCorrectly", MessageId = "Theming", Justification = "Correct spelling")]
 
-namespace Microsoft.Toolkit.Uwp.UI.Controls.Design.Common
+namespace Microsoft.Toolkit.Uwp.UI.Controls.Design
 {
+#if !VS_DESIGNER_PROCESS_ISOLATION
+    internal static partial class ControlTypes
+    {
+        // HACK: Don't forget to update, if the namespace changes.
+        private const string RootNamespace = "Microsoft.Toolkit.Uwp.UI.Controls";
```

Comment:
Anyway to do this without hardcoding it like this?

Reply:
I think that's part of the switch they made in VS is that everything is an indirect reference so they don't have to have a hard-link to the assembly?

I think the only other way we'd be able to keep a dependency for compilation but not for the binary would be to look into T4 templating or the experimental Source Generators. That may make it easier to automate some of this anyway? But I'm not sure how feasible that is?

Reply:
@Sergio0694 any ideas here?

@Nirmal4G is this the main thing left to figure out is if we can clean this up? Anything else we should be looking out for?

It'd be great actually if we could also add a `readme` somewhere in one of the main design folders that describes how this system works and what new component authors should know about adding metadata (or updating it) so that we can maintain this moving forward? @Kyaa-dost this type of thing makes me think we should have a few sub-docs to #3269 guidelines. This would include linking to things like the Sample App readme too.  Looking at the VSCode project they use the Wiki for all this developer documentation. Should we migrate to do something similar? @azchohfi @HerrickSpencer thoughts?

Reply:
If the documentation in a Readme.md starts getting long, it is time to split it up to more manageable pieces with links to separate sections. I'd say you should still start at the Readme.md and make links to sections in the Wiki. This is because very few will actually go directly to the Wiki. We've been trained to use the Readme.  
With this method you can have a nice concise Readme that provides a verbose amount of data in the wiki on each subject needed.  That wiki should also be written in 'bite-size' single subject pieces so as not to overwhelm the reader.  Use as many links/docs in the wiki as needed to make clear but thorough documentation. 
This is my 10¢.

Reply:
I propose `ReadMe.md` be the definitive source for these changes.

Reply:
> is this the main thing left to figure out is if we can clean this up?

Yep. This is all that's left.

I still have to add remaining metadata for all the controls but that'll be in another PR after 7.0 feature set is finalized.

There's still the Sdk-style migration pending for these projects. There are a few issues I'm hitting that I need to figure it out.

Reply:
@ranjeshj @chingucoding I noticed in [your PR here](https://github.com/microsoft/microsoft-ui-xaml/pull/3065) you just have a simple XML file. Is there part of your build system generating these metadata classes or something else? Or is that the piece you're missing to get the extra metadata for icons and other things like we're trying to update to here in the Toolkit?

Reply:
Afaik the WinUI nuget package does not do any meta class writing yet. The PR you linked only added missing controls to the designer though the XML file, however there are a few things that are currently missing for full designer support of WinUI, e.g. [NumberBox properties not being in the common category](https://github.com/microsoft/microsoft-ui-xaml/issues/1720)

Reply:
@Nirmal4G as it seems we don't plan to do into source code generating in this PR, I wonder if we can make a test that would break if the string does not match the namespace? If it's easy enough to add a test to this PR it could protect an error from sneaking past while it's still posible for the string to be invalid.

Reply:
I'll look into it. A test project for this is overkill. May be `Debug.Assert` could do the job.

I was looking into source generators before but there are some limitations that blocked me. I'll discuss those in my next PR. This one is about adding support for the new XAML Designer in VS.